Fix:

error: implicit declaration of function 'sendfile' is invalid in C99

https://github.com/ImageMagick/ImageMagick6/commit/3417f41c34b290ec97cafedfd9dc149eb708d727
https://github.com/ImageMagick/ImageMagick6/commit/6ecb84f4002051862fd87f002918818543769f39
--- configure.orig
+++ configure
@@ -22894,7 +22894,7 @@ fi
 
 
 # Check additional headers
-for ac_header in arm/limits.h arpa/inet.h complex.h errno.h fcntl.h limits.h linux/unistd.h locale.h machine/param.h mach-o/dyld.h netinet/in.h OS.h process.h sun_prefetch.h stdarg.h sys/ipc.h sys/mman.h sys/resource.h sys/sendfile.h sys/socket.h sys/syslimits.h sys/time.h sys/timeb.h sys/times.h sys/wait.h utime.h wchar.h xlocale.h
+for ac_header in arm/limits.h arpa/inet.h complex.h errno.h fcntl.h limits.h linux/unistd.h locale.h machine/param.h mach-o/dyld.h netinet/in.h OS.h process.h sun_prefetch.h stdarg.h sys/ipc.h sys/mman.h sys/resource.h sys/sendfile.h sys/socket.h sys/syslimits.h sys/time.h sys/timeb.h sys/times.h sys/uio.h sys/wait.h utime.h wchar.h xlocale.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
--- magick/cache.c.orig
+++ magick/cache.c
@@ -809,7 +809,7 @@ static MagickBooleanType ClonePixelCacheOnDisk(
   quantum=(size_t) MagickMaxBufferExtent;
   if ((fstat(cache_info->file,&file_stats) == 0) && (file_stats.st_size > 0))
     {
-#if defined(MAGICKCORE_HAVE_SENDFILE)
+#if defined(MAGICKCORE_HAVE_SENDFILE) && !defined(__APPLE__)
       if (cache_info->length < 0x7ffff000)
         {
           count=sendfile(clone_info->file,cache_info->file,(off_t *) NULL,
--- magick/studio.h.orig
+++ magick/studio.h
@@ -221,6 +221,12 @@ extern int vsnprintf(char *,size_t,const char *,va_list);
 # if defined(MAGICKCORE_HAVE_SYS_SENDFILE_H)
 #  include <sys/sendfile.h>
 # endif
+# if defined(MAGICKCORE_HAVE_SYS_SOCKET_H)
+#  include <sys/socket.h>
+# endif
+# if defined(MAGICKCORE_HAVE_SYS_UIO_H)
+#  include <sys/uio.h>
+# endif
 #endif
 #else
 # include <types.h>
